elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -56,14 +59,23 @@ public ElasticsearchSchemaFactory() {
 
     try {
       final String coordinatesString = (String) map.get("coordinates");
-      Preconditions.checkState(coordinatesString != null,
-          "'coordinates' is missing in configuration");
 
-      final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
-          new TypeReference<Map<String, Integer>>() { });
+      final String hostsString = (String) map.get("hosts");
```

Comment:
I prefer to composite this check like this:
```java
final List<String> hosts;
if (map.containsKey("coordinates")) {
  hosts = ...; // // build List<String> from coordinates
} else if (map.containsKey("hosts")) {
  hosts = ...; // build List<String> from hosts
} else {
   throw new IllegalArgumentException("Both 'coordinates' and 'hosts' is missing in configuration. Provide one of them.");
}
```

Reply:
Done